migration/create-environment.sh

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env bash
+set -ex
+
+ENVIRONMENT=$1
+
+# ----------------------- Cluster files ----------
+
+CLUSTER_DIRECTORY=clusters/$ENVIRONMENT
+
+mkdir -p $CLUSTER_DIRECTORY
+
+mkdir -p $CLUSTER_DIRECTORY/00
+mkdir -p $CLUSTER_DIRECTORY/01
+mkdir -p $CLUSTER_DIRECTORY/base
+
+cp k8s/$ENVIRONMENT/pub-cert.pem  $CLUSTER_DIRECTORY/pub-cert.pem
+
+(
+cat <<EOF
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+- ../../../apps/flux-system/sbox/base
+- ../../../apps/admin/base/kustomize.yaml
+- ../../../apps/kube-system/base/kustomize.yaml
+- ../../../apps/kured/base/kustomize.yaml
+- ../../../apps/monitoring/base/kustomize.yaml
+EOF
+) > "$CLUSTER_DIRECTORY/base/kustomization.yaml"
+  
+  
+(
+cat <<EOF
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+- ../base
+
+patchesStrategicMerge:
+- ../../../apps/flux-system/$ENVIRONMENT/00/kustomize.yaml
+EOF
+) > "$CLUSTER_DIRECTORY/00/kustomization.yaml"
+
+(
+cat <<EOF
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+- ../base
+
+patchesStrategicMerge:
+- ../../../apps/flux-system/$ENVIRONMENT/01/kustomize.yaml
+EOF
+) > "$CLUSTER_DIRECTORY/01/kustomization.yaml"
+  
+# ----------------------- Flux system files ----------
+
+FLUX_SYSTEM_DIRECTORY=apps/flux-system/$ENVIRONMENT
+
+mkdir -p $FLUX_SYSTEM_DIRECTORY
+
+mkdir -p $FLUX_SYSTEM_DIRECTORY/00
+mkdir -p $FLUX_SYSTEM_DIRECTORY/01
+mkdir -p $FLUX_SYSTEM_DIRECTORY/base
+
+(
+cat <<EOF
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - ../../base
+  - git-credentials.yaml
+EOF
+) > "$FLUX_SYSTEM_DIRECTORY/base/kustomization.yaml"
+
+(
+cat <<EOF
+apiVersion: kustomize.toolkit.fluxcd.io/v1beta1
+kind: Kustomization
+metadata:
+  name: flux-system
+  namespace: flux-system
+spec:
+  path: ./$CLUSTER_DIRECTORY/00
+  postBuild:
+    substitute:
+      ENVIRONMENT: "$ENVIRONMENT"
+      CLUSTER: "00"
+      ENV_MONITOR_CHANNEL: "aks-monitor-$ENVIRONMENT"
+      KEYVAULT_ENVIRONMENT: "$ENVIRONMENT"
+EOF
+) > "$FLUX_SYSTEM_DIRECTORY/00/kustomize.yaml"
+
+(
+cat <<EOF
+apiVersion: kustomize.toolkit.fluxcd.io/v1beta1
+kind: Kustomization
+metadata:
+  name: flux-system
+  namespace: flux-system
+spec:
+  path: ./$CLUSTER_DIRECTORY/00
+  postBuild:
+    substitute:
+      ENVIRONMENT: "$ENVIRONMENT"
+      CLUSTER: "01"
+      ENV_MONITOR_CHANNEL: "aks-monitor-$ENVIRONMENT"
+      KEYVAULT_ENVIRONMENT: "$ENVIRONMENT"
+EOF
+) > "$FLUX_SYSTEM_DIRECTORY/01/kustomize.yaml"
+
+
+kubectx cft-sbox-00-aks-admin
```

Comment:
maybe should be a variable at the top for the cluster name? and verify that kubectx is installed?

Reply:
change it to just relogin.